src/secure_mount.rs

```diff
@@ -18,36 +22,45 @@ use std::process::Command;
  * implementation as the original python version.
  */
 fn check_mount(secure_dir: &Path) -> Result<bool> {
-    let output = Command::new("mount").output()?;
-
-    let mount_result = String::from_utf8(output.stdout)?;
-
-    let lines: Vec<&str> = mount_result.split('\n').collect();
-
-    // Check mount list for secure directory
-    for line in lines {
-        let tokens: Vec<&str> = line.split(' ').collect();
-
-        if tokens.len() < 3 {
-            continue;
+    let f = fs::File::open(MOUNTINFO)?;
+    let f = BufReader::new(f);
+    let lines = f.lines();
+
+    for line in lines.flatten() {
+        if !line.contains('-') {
```

Comment:
Perhaps this check could be moved after splitting the fields, as filesystem type and volume name may also contain `-`.

Reply:
Oops, thanks for that, I wasn't paying attention. I changed the position (comparing to the original python implementation) to avoid splitting the string when it is not really parseable.

I think that checking for `-` with surrounding spaces is basically equivalent to check for `-` string after splitting on spaces. I'll try this.

Reply:
@ueno In the end I removed the initial "validation" as it wasn't really useful (let me know if you disagree). I changed the code to error in case the `-` separator is not found while parsing the lines.